datumaro/cli/contexts/model.py

```diff
@@ -46,6 +46,8 @@ def build_add_parser(parser_ctor=argparse.ArgumentParser):
 
     return parser
 
+on_error = None
```

Comment:
Having to declare a variable negates a half of the usability of this solution from my point of view. Maybe we can includea "default" name in `good-names` of `.pylintrc`? Probably, we could also rename `implicit` to `implicit_var` to make it more clear.

Reply:
How about we just dispense with the implicit variables and modify the semantics of `error_rollback` so that it can be used as follows:

```python
@error_rollback
def foobar():
    Rollback.current.add(my_rollback)
```

?

Reply:
I have no objections about such solution. But then we need either some class property (which can be done via metaclasses somehow), or we introduce a static function - with stack lookup (to obtain a function identifier). We need to have a separate rollback stacks for inner functions. It can be done even with a single global list (stack) of exit callbacks via some markers. Then, if we have an unhandled error, we execute functions (in the reverse order to the addition) until we meet a marker. Maybe I miss some elegant solution?

Reply:
I don't see why such complexity is needed. We can just set `Rollback.current` before the underlying function is called, and restore it back to the previous value once it's done. It's the same thing we do now, just with a class member variable instead of a module-level variable.

Reply:
Looks good. It is similar to what I described in the second half.